app/src/main/java/openfoodfacts/github/scrachx/openfood/features/scanhistory/ScanHistoryActivity.kt

```diff
@@ -340,9 +348,10 @@ class ScanHistoryActivity : BaseActivity(), SwipeController.Actions {
                         binding.historyProgressbar.visibility = View.GONE
                         binding.emptyHistoryInfo.visibility = View.VISIBLE
                         binding.scanFirst.visibility = View.VISIBLE
-                        invalidateOptionsMenu()
                         return@flatMapCompletable Completable.complete()
                     }
+                    showMenuButtons = true
```

Comment:
Could we remove this variable, keep the check inside onCreateOptionsMenu and just invalidate in on change?

Reply:
The problem is, initially every time when the view is created, the adapter is empty . So it is not possible to keep the check in the onCreateOptionsMenu, we need a variable for that.